### PR TITLE
[REF] call _get_product_accounts()

### PR DESCRIPTION
**Pull request by legalsylvain**
**Description of the issue/feature this PR addresses:**
this PR is a refactor.
the product has a new function ``_get_product_accounts``, but this function is not called in all Odoo.

This PR fixes that.

kind regards.

CC : @mart-e 

CC : @OCA/accounting-maintainers

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by feketemihai** [comment]
@legalsylvain It was used last time in version 11.

**Comment by feketemihai** [comment]
@legalsylvain I would use it since it will make easier inheritance, and don't have to change in so many places.

**Comment by ivantodorovich** [comment]
:+1:

**Comment by legalsylvain** [comment]
@mart-e : 

- I don't think it's a ``point-of-sale`` PR. It's more a fix, because Odoo in many modules, doesn't call a new ``_get_product_accounts``.
- Could you say me if there any chance to have this PR reviewed and merged ? just to know if I finish it. For the time being, I just changed 2 calls, but they are a lot of errors in Odoo code in master.

thanks !

**Comment by smetl** [comment]
@legalsylvain It seems to be a good idea :)

**Comment by legalsylvain** [comment]
@smetl : So if I finish my PR, Can I expect it will be merge ? 

thanks for your answer.

**Comment by smetl** [comment]
@legalsylvain Can you squash your commits plz?

**Comment by legalsylvain** [comment]
@smetl : 

1) commits squashed.
2) PR finished. same fix applied in ``point_of_sale``, ``stock_account`` and ``sale`` module.

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -600,7 +600,7 @@ def _prepare_line(self, order_line):
         """
         def get_income_account(order_line):
             product = order_line.product_id
-            income_account = product.with_company(order_line.company_id).property_account_income_id or product.categ_id.with_company(order_line.company_id).property_account_income_categ_id
+            income_account = product.with_company(order_line.company_id)._get_product_accounts['income']
```
Comment (smetl) [inline addons/point_of_sale/models/pos_session.py]: missing parenthesis

**Comment by smetl** [comment]
@robodoo r+

**Comment by legalsylvain** [comment]
Thanks !

**Pull request closed without merging**